source/common/config/well_known_names.h

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "envoy/common/exception.h"
+
+#include "common/common/singleton.h"
+
+#include "fmt/format.h"
+
+namespace Envoy {
+namespace Config {
+
+/**
+ * Converts certain names from v1 to v2 by adding a prefix.
+ */
+class V1Converter {
+public:
+  /**
+   * @param v2_names vector of all the v2 names that may be converted.
+   */
+  V1Converter(const std::vector<std::string>& v2_names) {
+    const std::string prefix = "envoy.";
+    for (const auto& name : v2_names) {
+      // Ensure there are no misplaced names provided to this constructor.
+      if (name.find(prefix) != 0) {
+        throw EnvoyException(fmt::format(
+            "Attempted to create a conversion for a v2 name that isn't prefixed by {}", prefix));
+      }
+      v1_to_v2_names_[name.substr(prefix.size())] = name;
```

Comment:
I think this could be done a bit simpler by just prefixing on ingest, but it seems alright, and will eventually be deprecated.